File: docs/JAMIA/comparisons/RPython/TM/25937000_geom_polygon_discrete_color.ts
/**
 * RPython SO#25937000 — geom_polygon color=factor on continuous color scale
 * Effect: Crash
 * Bug class: Type coercion
 *
 * R bug: One layer uses continuous color (numeric diff), another uses discrete
 * microstyle (character). Mixing crashes: "Continuous value supplied to discrete scale."
 *
 * In tidy-ts, a string microstyle column cannot be used in numeric operations.
 * The type system prevents mixing discrete string values with continuous numeric scales.
 */
import { createDataFrame, stats as s } from "@tidy-ts/dataframe";

const splineHull = createDataFrame([
  { pauseMedian: 1.2, numTotalPauses: 8, microstyle: "staccato" },
  { pauseMedian: 2.4, numTotalPauses: 14, microstyle: "legato" },
  { pauseMedian: 0.8, numTotalPauses: 5, microstyle: "staccato" },
]);

// The SO user's intent: use microstyle as a color scale alongside numeric diff.
// ggplot2 crashes because discrete and continuous color are mixed.
// tidy-ts: correlating string microstyle with numeric values requires number[].
// @ts-expect-error — string[] is not assignable to number[]
s.test.correlation.pearson({ x: splineHull.extract("microstyle"), y: splineHull.extract("pauseMedian") });

File: docs/JAMIA/comparisons/RPython/TM/25937000_geom_polygon_discrete_color.R
# ID: SO#25937000
# Language: R
# Bug class: Value type
# Runtime consequence: Crash
# In study: Yes
# Inclusion rationale: String/factor value on continuous scale. Same pattern as 29278153.
# Reproduction status: Live
# Type system catch: Continuous scale requires `number`; `string | factor` rejected

library(ggplot2)

merged <- data.frame(
  pauseMedian = c(1, 2, 3, 4, 5),
  numTotalPauses = c(10, 12, 11, 13, 9),
  diff = c(0.1, 0.2, 0.3, 0.4, 0.5)
)

splineHull <- data.frame(
  pauseMedian = c(1, 2, 3, 1),
  numTotalPauses = c(8, 14, 10, 8),
  microstyle = factor(c("A", "A", "B", "B"))
)

ggplot(data = merged, aes(x = pauseMedian, y = numTotalPauses, color = diff)) +
  geom_point() +
  geom_polygon(
    data = splineHull,
    mapping = aes(
      x = pauseMedian,
      y = numTotalPauses,
      group = microstyle,
      color = microstyle
    ),
    alpha = 0
  )
